Java Edition 1.20.5

1.20.5 is currently in snapshot phase. I expect sgui to break because there are changes to networking code.

- [x] Upstream: Java Edition 1.20.5
- [x] Upstream: Fabric Loader for 1.20.5
- [x] Upstream: Fabric API for 1.20.5
- [x] Upstream: Patbox's sgui for 1.20.5
- Update Stonks to 1.20.5
  + [x] Deprecate Nahara Toolkit Tasks
  + [x] Remove Nahara Modkit Annotations and move back to manual JSON editing
  + [x] Update default configuration file to use components
  + [x] Write changelogs
  + [x] Test

```
java.lang.RuntimeException: Error creating Mixin config fabric-game-rule-api-v1.mixins.json for mod fabric-game-rule-api-v1
	at net.fabricmc.loader.impl.launch.FabricMixinBootstrap.init(FabricMixinBootstrap.java:96) ~[fabric-loader-0.15.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:151) ~[fabric-loader-0.15.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:68) ~[fabric-loader-0.15.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) ~[fabric-loader-0.15.10.jar:?]
	at net.fabricmc.devlaunchinjector.Main.main(Main.java:86) ~[?:?]
Caused by: java.lang.NoSuchFieldError: Class org.spongepowered.asm.mixin.MixinEnvironment$CompatibilityLevel does not have member field 'org.spongepowered.asm.mixin.MixinEnvironment$CompatibilityLevel JAVA_22'
	at net.fabricmc.loader.impl.launch.knot.MixinServiceKnot.getMaxCompatibilityLevel(MixinServiceKnot.java:227) ~[fabric-loader-0.15.10.jar:?]
	at org.spongepowered.asm.mixin.MixinEnvironment.setCompatibilityLevel(MixinEnvironment.java:1577) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinConfig.initCompatibilityLevel(MixinConfig.java:557) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinConfig.postInit(MixinConfig.java:502) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinConfig.onLoad(MixinConfig.java:430) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinConfig.create(MixinConfig.java:1293) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.Config.create(Config.java:148) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.Mixins.createConfiguration(Mixins.java:100) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.Mixins.addConfiguration(Mixins.java:87) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at net.fabricmc.loader.impl.launch.FabricMixinBootstrap.init(FabricMixinBootstrap.java:94) ~[fabric-loader-0.15.10.jar:?]
	... 4 more
```

Am I doing something wrong?

--- a/.github/workflows/gradle.yml
+++ b/.github/workflows/gradle.yml
@@ -15,10 +15,10 @@ jobs:
 
     steps:
     - uses: actions/checkout@v3
-    - name: Set up JDK 17
+    - name: Set up JDK 21
       uses: actions/setup-java@v3
       with:
-        java-version: '17'
+        java-version: '21'
         distribution: 'temurin'
     - name: Build with Gradle
       uses: gradle/gradle-build-action@v2.6.0

--- a/.github/workflows/gradle.yml
+++ b/.github/workflows/gradle.yml
@@ -14,9 +14,9 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
     - name: Set up JDK 21
-      uses: actions/setup-java@v3
+      uses: actions/setup-java@v4
       with:
         java-version: '21'
         distribution: 'temurin'
@@ -24,7 +24,7 @@ jobs:
       uses: gradle/gradle-build-action@v2.6.0
       with:
         arguments: build
-    - uses: actions/upload-artifact@v3
+    - uses: actions/upload-artifact@v4
       with:
         name: Fabric Artifacts
         path: fabric/build/libs/

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -12,9 +12,9 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - name: Set up JDK 21
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
         with:
           java-version: '21'
           distribution: 'temurin'
@@ -31,7 +31,7 @@ jobs:
             -Pstonks_version=${{ steps.tag.outputs.tag }}
             --stacktrace
             --warning-mode=fail
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
         with:
           name: "stonks-fabric-${{ steps.tag.outputs.tag }}-release"
           path: fabric/build/libs/

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -13,10 +13,10 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - name: Set up JDK 17
+      - name: Set up JDK 21
         uses: actions/setup-java@v3
         with:
-          java-version: '17'
+          java-version: '21'
           distribution: 'temurin'
       - name: Get current tag
         id: tag

--- a/README.md
+++ b/README.md
@@ -4,7 +4,7 @@
 ![Build Status Badge](https://img.shields.io/github/actions/workflow/status/nahkd123/stonks/gradle.yml)
 ![License Badge](https://img.shields.io/github/license/nahkd123/stonks)
 ![Dynamic Version Badge](https://img.shields.io/github/v/release/nahkd123/stonks)
-![Minecraft Version Badge](https://img.shields.io/badge/minecraft-1.20.1-red)
+![Minecraft Version Badge](https://img.shields.io/badge/minecraft-1.20.5-red)
 
 Weird little stocks/items market for block game.
 

--- a/build.gradle
+++ b/build.gradle
@@ -41,7 +41,7 @@ subprojects {
         withSourcesJar()
 
         toolchain {
-            languageVersion = JavaLanguageVersion.of(17)
+            languageVersion = JavaLanguageVersion.of(21)
         }
     }
 

--- a/fabric/build.gradle
+++ b/fabric/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-    id 'fabric-loom' version '1.4-SNAPSHOT'
+    id 'fabric-loom' version '1.6-SNAPSHOT'
     id 'com.diffplug.eclipse.apt' version '3.42.2'
 }
 
@@ -7,20 +7,12 @@ repositories {
     maven { url = 'https://maven.nucleoid.xyz' }
 }
 
-eclipse {
-    synchronizationTasks 'eclipseFactorypath'
-}
+processResources {
+    inputs.property "stonks_version", rootProject.stonks_version
 
-compileJava {
-    def expands = Map.of(
-        "stonks_version", rootProject.stonks_version,
-        "minecraft_version", rootProject.minecraft_version,
-        "loader_version", rootProject.loader_version,
-        "fabric_version", rootProject.fabric_version
-    ).entrySet().stream()
-    .map(v -> "${v.getKey()}:${v.getValue()}")
-    .collect(java.util.stream.Collectors.joining(';'))
-    options.compilerArgs << "-Anahara.modkit.expand=${expands}"
+    filesMatching("fabric.mod.json") {
+        expand "stonks_version": rootProject.stonks_version
+    }
 }
 
 dependencies {
@@ -28,10 +20,11 @@ dependencies {
     mappings "net.fabricmc:yarn:${rootProject.yarn_mappings}:v2"
     modApi "net.fabricmc:fabric-loader:${rootProject.loader_version}"
 
-    implementation annotationProcessor(naharaModkit('nahara-modkit-annotations-v1'))
     modImplementation "net.fabricmc.fabric-api:fabric-api:$rootProject.fabric_version"
     modImplementation include("eu.pb4:sgui:$rootProject.sgui_version")
-    modImplementation include("eu.pb4:common-economy-api:$rootProject.common_economy_version")
+    modImplementation include("eu.pb4:common-economy-api:$rootProject.common_economy_version") {
+        exclude group: 'net.fabricmc'
+    }
     api include(project(':core'))
     api include(naharaToolkit('nahara-common-tasks'))
     api include(naharaToolkit('nahara-common-configurations'))

--- a/fabric/src/main/java/stonks/fabric/StonksFabric.java
+++ b/fabric/src/main/java/stonks/fabric/StonksFabric.java
@@ -31,14 +31,11 @@ import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import nahara.common.configurations.Config;
-import nahara.modkit.annotations.v1.Dependencies;
-import nahara.modkit.annotations.v1.Dependency;
-import nahara.modkit.annotations.v1.EntryPoint;
-import nahara.modkit.annotations.v1.Mod;
 import net.fabricmc.fabric.api.command.v2.CommandRegistrationCallback;
 import net.fabricmc.fabric.api.event.lifecycle.v1.ServerLifecycleEvents;
 import net.fabricmc.fabric.api.event.lifecycle.v1.ServerTickEvents;
 import net.fabricmc.loader.api.FabricLoader;
+import net.minecraft.component.DataComponentTypes;
 import net.minecraft.item.ItemStack;
 import net.minecraft.item.Items;
 import net.minecraft.server.MinecraftServer;
@@ -61,24 +58,11 @@ import stonks.fabric.service.IntegratedStonksService;
 import stonks.fabric.service.IntegratedUnstableStonksService;
 import stonks.fabric.service.StonksServiceProvider;
 
-@Mod(
-	modid = StonksFabric.MODID,
-	name = "Stonks2",
-	version = "${stonks_version}",
-	description = "Stonks2 for Fabric",
-	authors = "nahkd123",
-	license = "MIT")
-@Dependencies({
-	@Dependency(value = "fabricloader", version = ">=${loader_version}"),
-	@Dependency(value = "minecraft", version = ">=${minecraft_version}"),
-	@Dependency(value = "java", version = ">=17"),
-	@Dependency(value = "fabric-api", version = ">=${fabric_version}")
-})
 public class StonksFabric {
 	public static final String MODID = "stonks";
 	public static final Logger LOGGER = LoggerFactory.getLogger(MODID);
 
-	@EntryPoint
+	// Entry point
 	public static void init() {
 		LOGGER.info("Stonks2 is now initializing...");
 		LOGGER.info("Configuration directory is '{}'", getConfigDir());
@@ -175,7 +159,7 @@ public class StonksFabric {
 
 		if (out == null) {
 			out = new ItemStack(Items.BARRIER);
-			out.setCustomName(Text.literal(product.getProductName() + " (Invaild display)")
+			out.set(DataComponentTypes.ITEM_NAME, Text.literal(product.getProductName() + " (Invaild display)")
 				.styled(s -> s.withColor(Formatting.RED)));
 		}
 

--- a/fabric/src/main/java/stonks/fabric/StonksFabricUtils.java
+++ b/fabric/src/main/java/stonks/fabric/StonksFabricUtils.java
@@ -22,12 +22,8 @@
 package stonks.fabric;
 
 import java.text.DecimalFormat;
-import java.util.Objects;
 import java.util.Optional;
 
-import net.minecraft.item.ItemStack;
-import net.minecraft.registry.RegistryKeys;
-import net.minecraft.server.MinecraftServer;
 import net.minecraft.text.MutableText;
 import net.minecraft.text.Text;
 import net.minecraft.util.Formatting;
@@ -61,17 +57,6 @@ public class StonksFabricUtils {
 		return Text.literal(s);
 	}
 
-	public static String createStringFor(ItemStack stack, MinecraftServer server) {
-		var reg = server.getRegistryManager().getOptional(RegistryKeys.ITEM);
-		if (reg.isEmpty()) return null;
-		var key = reg.get().getKey(stack.getItem()).map(v -> v.getValue().toString());
-		if (key.isEmpty()) return null;
-
-		var out = key.get();
-		if (stack.hasNbt()) out += stack.getNbt().asString();
-		return out;
-	}
-
 	public static final DecimalFormat CURRENCY_FORMATTER = new DecimalFormat("#,##0.##");
 	public static final DecimalFormat TAX_FORMATTER = new DecimalFormat("#,##0.##%");
 
@@ -93,16 +78,5 @@ public class StonksFabricUtils {
 			: Translations.Messages.OfferInfoText$Sell;
 		return Translations.Messages.OfferInfoText(typeText, offer.totalAvailableUnits(), offer.offers(),
 			offer.pricePerUnit());
-	}
-
-	public static boolean compareStack(ItemStack a, ItemStack b) {
-		if (a.isEmpty()) return b.isEmpty();
-		if (b.isEmpty()) return a.isEmpty();
-		if (!a.isOf(b.getItem())) return false;
-		var nbtA = a.getNbt();
-		if (nbtA != null && nbtA.isEmpty()) nbtA = null;
-		var nbtB = b.getNbt();
-		if (nbtB != null && nbtB.isEmpty()) nbtB = null;
-		return Objects.equals(nbtA, nbtB);
 	}
 }

--- a/fabric/src/main/java/stonks/fabric/adapter/provided/ItemsAdapter.java
+++ b/fabric/src/main/java/stonks/fabric/adapter/provided/ItemsAdapter.java
@@ -34,7 +34,6 @@ import net.minecraft.server.command.CommandManager;
 import net.minecraft.server.network.ServerPlayerEntity;
 import stonks.core.product.Product;
 import stonks.fabric.StonksFabric;
-import stonks.fabric.StonksFabricUtils;
 import stonks.fabric.adapter.StonksFabricAdapter;
 import stonks.fabric.provider.StonksProvidersRegistry;
 
@@ -54,7 +53,7 @@ public class ItemsAdapter implements StonksFabricAdapter {
 		var str = product.getProductConstructionData();
 		if (str == null) return null;
 
-		// String format: "item [namespace:]<id>[{<nbt>}]"
+		// String format: "item [namespace:]<id>[<component>][{<nbt>}]"
 		if (str.startsWith(PREFIX)) {
 			str = str.substring(PREFIX.length());
 
@@ -88,7 +87,7 @@ public class ItemsAdapter implements StonksFabricAdapter {
 
 		for (int i = 0; i < inv.size(); i++) {
 			var stack = inv.getStack(i);
-			if (stack.isEmpty() || !StonksFabricUtils.compareStack(refStack, stack)) continue;
+			if (stack.isEmpty() || !ItemStack.areEqual(refStack, stack)) continue;
 			count += stack.getCount();
 		}
 
@@ -103,7 +102,6 @@ public class ItemsAdapter implements StonksFabricAdapter {
 
 		var inv = player.getInventory();
 		var giveStack = refStack.copyWithCount(amount);
-		if (!giveStack.hasNbt() || giveStack.getNbt().isEmpty()) giveStack.setNbt(null); // Fix stacking issue
 
 		if (!inv.insertStack(giveStack)) {
 			var e = player.dropItem(giveStack, false);
@@ -124,7 +122,7 @@ public class ItemsAdapter implements StonksFabricAdapter {
 
 		for (int i = 0; i < inv.size(); i++) {
 			var stack = inv.getStack(i);
-			if (stack.isEmpty() || !StonksFabricUtils.compareStack(refStack, stack)) continue;
+			if (stack.isEmpty() || !ItemStack.areEqual(refStack, stack)) continue;
 
 			var toTake = Math.min(amount, stack.getCount());
 			stack.decrement(toTake);

--- a/fabric/src/main/java/stonks/fabric/adapter/provided/ItemsAdapter.java
+++ b/fabric/src/main/java/stonks/fabric/adapter/provided/ItemsAdapter.java
@@ -87,7 +87,7 @@ public class ItemsAdapter implements StonksFabricAdapter {
 
 		for (int i = 0; i < inv.size(); i++) {
 			var stack = inv.getStack(i);
-			if (stack.isEmpty() || !ItemStack.areEqual(refStack, stack)) continue;
+			if (stack.isEmpty() || !ItemStack.areItemsAndComponentsEqual(refStack, stack)) continue;
 			count += stack.getCount();
 		}
 
@@ -122,7 +122,7 @@ public class ItemsAdapter implements StonksFabricAdapter {
 
 		for (int i = 0; i < inv.size(); i++) {
 			var stack = inv.getStack(i);
-			if (stack.isEmpty() || !ItemStack.areEqual(refStack, stack)) continue;
+			if (stack.isEmpty() || !ItemStack.areItemsAndComponentsEqual(refStack, stack)) continue;
 
 			var toTake = Math.min(amount, stack.getCount());
 			stack.decrement(toTake);

--- a/fabric/src/main/java/stonks/fabric/adapter/provided/ScoreboardEconomyAdapter.java
+++ b/fabric/src/main/java/stonks/fabric/adapter/provided/ScoreboardEconomyAdapter.java
@@ -21,6 +21,7 @@
  */
 package stonks.fabric.adapter.provided;
 
+import net.minecraft.scoreboard.ReadableScoreboardScore;
 import net.minecraft.scoreboard.Scoreboard;
 import net.minecraft.scoreboard.ScoreboardCriterion;
 import net.minecraft.scoreboard.ScoreboardCriterion.RenderType;
@@ -69,7 +70,8 @@ public class ScoreboardEconomyAdapter implements StonksFabricAdapter {
 
 	@Override
 	public double accountBalance(ServerPlayerEntity player) {
-		return scoreToMoney(scoreboard.getScore(player, getObjective()).getScore());
+		ReadableScoreboardScore scoreEntry = scoreboard.getScore(player, getObjective());
+		return scoreToMoney(scoreEntry != null ? scoreEntry.getScore() : 0);
 	}
 
 	@Override

--- a/fabric/src/main/java/stonks/fabric/menu/MarketMainMenu.java
+++ b/fabric/src/main/java/stonks/fabric/menu/MarketMainMenu.java
@@ -228,7 +228,7 @@ public class MarketMainMenu extends StackedMenu {
 				return GuiElementBuilder.from(dispStack)
 					.setLore(new ArrayList<>()) // Clear all lore
 					.addLoreLine(Text.literal(category.getCategoryName())
-						.styled(s -> s.withColor(Formatting.DARK_GRAY)))
+						.styled(s -> s.withColor(Formatting.DARK_GRAY).withItalic(false)))
 					.addLoreLine(Text.empty())
 					.addLoreLine(Translations.Menus.MainMenu.product$instantBuy(instantBuyPrice))
 					.addLoreLine(Translations.Menus.MainMenu.product$instantSell(instantSellPrice))

--- a/fabric/src/main/java/stonks/fabric/menu/MenuIcons.java
+++ b/fabric/src/main/java/stonks/fabric/menu/MenuIcons.java
@@ -28,7 +28,7 @@ import stonks.fabric.translation.Translations;
 
 public class MenuIcons {
 	public static final GuiElementBuilder BORDER = new GuiElementBuilder(Items.BLACK_STAINED_GLASS_PANE)
-		.setName(Translations.Icons.Border);
+		.hideTooltip();
 
 	public static final GuiElementBuilder BACK = new GuiElementBuilder(Items.ARROW)
 		.setName(Translations.Icons.PreviousMenu)

--- a/fabric/src/main/java/stonks/fabric/menu/product/OfferAmountConfigureMenu.java
+++ b/fabric/src/main/java/stonks/fabric/menu/product/OfferAmountConfigureMenu.java
@@ -68,7 +68,8 @@ public class OfferAmountConfigureMenu extends StackedMenu {
 
 		setSlot(25, new GuiElementBuilder(Items.DARK_OAK_SIGN)
 			.setName(Translations.Menus.CreateOffer.CustomAmount)
-			.addLoreLine(Text.literal(product.getProductName()).styled(s -> s.withColor(Formatting.GRAY)))
+			.addLoreLine(Text.literal(product.getProductName())
+				.styled(s -> s.withColor(Formatting.GRAY).withItalic(false)))
 			.addLoreLine(Text.empty())
 			.addLoreLine(Translations.Menus.CreateOffer.ClickForAmount)
 			.setCallback((index, type, action, gui) -> new OfferSelectCustomAmountInput(player, this).open()));
@@ -102,7 +103,8 @@ public class OfferAmountConfigureMenu extends StackedMenu {
 
 		return new GuiElementBuilder(disabled ? Items.BARRIER : icon, Math.min(Math.max(amount / 64, 1), 64))
 			.setName(buttonName)
-			.addLoreLine(Text.literal(product.getProductName()).styled(s -> s.withColor(Formatting.GRAY)))
+			.addLoreLine(Text.literal(product.getProductName())
+				.styled(s -> s.withColor(Formatting.GRAY).withItalic(false)))
 			.addLoreLine(Text.empty())
 			.addLoreLine(disabled
 				? Translations.Menus.CreateOffer.NoOfferForYou

--- a/fabric/src/main/java/stonks/fabric/misc/StonksSounds.java
+++ b/fabric/src/main/java/stonks/fabric/misc/StonksSounds.java
@@ -61,7 +61,7 @@ public class StonksSounds {
 			var e = iter.next();
 
 			if (e.atTime <= soundTime) {
-				e.player.playSound(e.sound, SoundCategory.PLAYERS, e.volume, e.pitch);
+				e.player.playSoundToPlayer(e.sound, SoundCategory.PLAYERS, e.volume, e.pitch);
 				iter.remove();
 			}
 		}
@@ -70,7 +70,7 @@ public class StonksSounds {
 	}
 
 	public void play(ServerPlayerEntity player, SoundEvent sound, int ticks, float volume, float pitch) {
-		if (ticks <= 0) player.playSound(sound, SoundCategory.PLAYERS, volume, pitch);
+		if (ticks <= 0) player.playSoundToPlayer(sound, SoundCategory.PLAYERS, volume, pitch);
 		else entries.add(new Entry(player, sound, soundTime + ticks, volume, pitch));
 	}
 

--- a/fabric/src/main/java/stonks/fabric/mixin/MinecraftServerMixin.java
+++ b/fabric/src/main/java/stonks/fabric/mixin/MinecraftServerMixin.java
@@ -26,7 +26,6 @@ import java.util.List;
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.Unique;
 
-import nahara.modkit.annotations.v1.AutoMixin;
 import net.minecraft.server.MinecraftServer;
 import stonks.core.caching.StonksServiceCache;
 import stonks.core.service.LocalStonksService;
@@ -42,7 +41,6 @@ import stonks.fabric.misc.StonksSounds;
 import stonks.fabric.misc.TasksHandler;
 import stonks.fabric.service.StonksServiceProvider;
 
-@AutoMixin
 @Mixin(MinecraftServer.class)
 public abstract class MinecraftServerMixin implements StonksFabricPlatform {
 	@Unique

--- a/fabric/src/main/java/stonks/fabric/service/IntegratedStonksService.java
+++ b/fabric/src/main/java/stonks/fabric/service/IntegratedStonksService.java
@@ -72,12 +72,14 @@ public class IntegratedStonksService extends StonksMemoryService {
 		}
 
 		try (var stream = Files.newInputStream(saveFilePath)) {
-			var offer = Offer.deserialize(this::productGetter, stream);
-			if (offer != null) insertOffer(offer);
-			else {
-				StonksFabric.LOGGER.info("Loaded data from {}", saveFilePath);
-				return;
-			}
+			Offer offer;
+
+			do {
+				offer = Offer.deserialize(this::productGetter, stream);
+				if (offer != null) insertOffer(offer);
+			} while (offer != null);
+
+			StonksFabric.LOGGER.info("Loaded data from {}", saveFilePath);
 		} catch (IOException e) {
 			e.printStackTrace();
 			StonksFabric.LOGGER.error("Unable to load data from {}", saveFilePath);

--- a/fabric/src/main/java/stonks/fabric/translation/Translations.java
+++ b/fabric/src/main/java/stonks/fabric/translation/Translations.java
@@ -36,6 +36,7 @@ import stonks.fabric.StonksFabricUtils;
 public final class Translations {
 	// @formatter:off
 	public static final class Icons {
+		@Deprecated(forRemoval = true)
 		public static final Text Border = translatableWithFallback("stonks.menu.icon.border", " ");
 		public static final Text PreviousMenu = translatableWithFallback("stonks.menu.icon.previousMenu", "\u00a77<-- \u00a7aBack");
 		public static final Text MainMenu = translatableWithFallback("stonks.menu.icon.mainMenu", "\u00a7eMarket menu");
@@ -77,13 +78,13 @@ public final class Translations {
 			public static final Text Offer$Buy(Offer offer) { return translatableWithFallback("stonks.menu.viewOffers.offer.buy", "\u00a7a\u00a7lBUY OFFER: %s", productName(offer.getProduct())); }
 			public static final Text Offer$Sell(Offer offer) { return translatableWithFallback("stonks.menu.viewOffers.offer.sell", "\u00a7e\u00a7lSELL OFFER: %s", productName(offer.getProduct())); }
 			public static final Text Offer$Progress(Offer offer) {
-				return translatableWithFallback("stonks.menu.viewOffers.offer.progress", "\u00a77Progress: %s %s/%s/%s",
+				return translatableWithFallback("stonks.menu.viewOffers.offer.progress", "\u00a77Progress: %s %s\u00a77/%s\u00a77/%s",
 					StonksFabricUtils.progressBar(20, Formatting.DARK_GRAY,
 						new double[] { offer.getClaimedUnits() / (double) offer.getTotalUnits(), offer.getFilledUnits() / (double) offer.getTotalUnits() },
 						new Formatting[] { Formatting.GREEN, Formatting.YELLOW }),
-					Text.literal(Integer.toString(offer.getClaimedUnits())).styled(s -> s.withColor(Formatting.GREEN)),
-					Text.literal(Integer.toString(offer.getFilledUnits())).styled(s -> s.withColor(Formatting.YELLOW)),
-					Text.literal(Integer.toString(offer.getTotalUnits())).styled(s -> s.withColor(Formatting.WHITE)));
+					Text.literal(Integer.toString(offer.getClaimedUnits())).styled(s -> s.withColor(Formatting.GREEN).withItalic(false)),
+					Text.literal(Integer.toString(offer.getFilledUnits())).styled(s -> s.withColor(Formatting.YELLOW).withItalic(false)),
+					Text.literal(Integer.toString(offer.getTotalUnits())).styled(s -> s.withColor(Formatting.WHITE).withItalic(false)));
 			}
 			public static final Text Offer$PricePerUnit(Offer offer) { return translatableWithFallback("stonks.menu.viewOffers.offer.pricePerUnit", "\u00a77Price per unit: %s", currency(offer.getPricePerUnit())); }
 			public static final Text Offer$TotalPrice(Offer offer) { return translatableWithFallback("stonks.menu.viewOffers.offer.totalPrice", "\u00a77Total price: %s", currency(offer.getPricePerUnit() * offer.getTotalUnits())); }
@@ -212,7 +213,7 @@ public final class Translations {
 		public static final Text SoldWithExtras(int amount, Product product, double moneySpend, int unitsLeft) { return translatableWithFallback("stonks.message.soldWithExtras", "Sold %sx %s for %s with %s units can't be sold", units(amount), productName(product), currency(moneySpend), units(unitsLeft)); }
 		public static final Text PlacedBuyOffer(int units, Product product, double totalPrice, double pricePerUnit) { return translatableWithFallback("stonks.message.placedBuyOffer", "Placed buy offer: %sx %s for %s @ %s/each", units(units), productName(product), currency(totalPrice), currency(pricePerUnit)); }
 		public static final Text PlacedSellOffer(int units, Product product, double totalPrice, double pricePerUnit) { return translatableWithFallback("stonks.message.placedSellOffer", "Placed sell offer: %sx %s for %s @ %s/each", units(units), productName(product), currency(totalPrice), currency(pricePerUnit)); }
-		public static final Text Currency(double val) { return translatableWithFallback("stonks.message.currency", "\u00a7e$%s", Text.literal(StonksFabricUtils.CURRENCY_FORMATTER.format(val)).styled(s -> s.withColor(Formatting.YELLOW))); }
+		public static final Text Currency(double val) { return translatableWithFallback("stonks.message.currency", "\u00a7e$%s", Text.literal(StonksFabricUtils.CURRENCY_FORMATTER.format(val)).styled(s -> s.withColor(Formatting.YELLOW).withItalic(false))); }
 		public static final Text OfferInfoText(Text typeText, int totalAvailable, int offers, double pricePerUnit) { return translatableWithFallback("stonks.message.offerInfoText", "\u00a77%s %s\u00a77x from %s \u00a77offers for %s\u00a77/each", typeText, units(totalAvailable), units(offers), currency(pricePerUnit)); }
 		public static final Text BuyOfferFilled(Offer offer) { return translatableWithFallback("stonks.message.buyOfferFilled", "\u00a77\u00a7l[\u00a7eStonks\u00a77\u00a7l]\u00a7r Your %sx %s buy offer has been filled!", units(offer.getTotalUnits()), productName(offer.getProduct())); }
 		public static final Text SellOfferFilled(Offer offer) { return translatableWithFallback("stonks.message.sellOfferFilled", "\u00a77\u00a7l[\u00a7eStonks\u00a77\u00a7l]\u00a7r Your %sx %s sell offer has been filled!", units(offer.getTotalUnits()), productName(offer.getProduct())); }
@@ -220,11 +221,11 @@ public final class Translations {
 	// @formatter:on
 
 	public static Text units(int units) {
-		return Text.literal(Integer.toString(units)).styled(s -> s.withColor(Formatting.AQUA));
+		return Text.literal(Integer.toString(units)).styled(s -> s.withColor(Formatting.AQUA).withItalic(false));
 	}
 
 	public static Text productName(Product product) {
-		return Text.literal(product.getProductName()).styled(s -> s.withColor(Formatting.AQUA));
+		return Text.literal(product.getProductName()).styled(s -> s.withColor(Formatting.AQUA).withItalic(false));
 	}
 
 	public static Text currency(double value) {

--- a/fabric/src/main/resources/default-config
+++ b/fabric/src/main/resources/default-config
@@ -53,14 +53,12 @@ useService stonks.fabric.service.IntegratedStonksService
         name Special Items
         product special_paper
             name Special Paper
-            construction item minecraft:paper{ \
-                display: { \
-                    Name: '{"text": "Special Paper", "color": "red", "italic": false}', \
-                    Lore: [ \
-                        '{"text": "Edit this in /config/stonks/config!", "color": "aqua", "italic": false}' \
-                    ] \
-                } \
-            }
+            construction item minecraft:paper[ \
+                minecraft:item_name={"text": "Special Paper", "color": "red", "italic": false}, \
+                minecraft:lore=[ \
+                    '{"text": "Edit this in /config/stonks/config!", "color": "aqua", "italic": false}' \
+                ] \
+            ]
 
 // Adapters
 useAdapter stonks.fabric.adapter.provided.ItemsAdapter

--- a/fabric/src/main/resources/fabric.mod.json
+++ b/fabric/src/main/resources/fabric.mod.json
@@ -1,0 +1,23 @@
+{
+    "schemaVersion": 1,
+    "id": "stonks",
+    "name": "Stonks2",
+    "version": "${stonks_version}",
+    "description": "Stonks2 for Fabric",
+    "environment": "*",
+    "license": "MIT",
+    "icon": "assets/stonks/icon.png",
+    "authors": ["nahkd123"],
+    "entrypoints": {
+        "main": ["stonks.fabric.StonksFabric::init"],
+        "client": [],
+        "server": []
+    },
+    "mixins": ["stonks.mixins.json"],
+    "depends": {
+        "fabricloader": ">=0.15.10",
+        "minecraft": ">=1.20.5",
+        "java": ">=21",
+        "fabric-api": "*"
+    }
+}

--- a/fabric/src/main/resources/stonks.mixins.json
+++ b/fabric/src/main/resources/stonks.mixins.json
@@ -1,0 +1,11 @@
+{
+    "required": true,
+    "package": "stonks.fabric.mixin",
+    "compatibilityLevel": "JAVA_21",
+    "mixins": [
+        "MinecraftServerMixin"
+    ],
+    "injectors": {
+        "defaultRequire": 1
+    }
+}

--- a/gradle.properties
+++ b/gradle.properties
@@ -1,16 +1,15 @@
 org.gradle.jvmargs=-Xmx1G
-org.gradle.parallel=true
 
 # Stonks
-stonks_version=2.0.0+1.20.3
+stonks_version=2.0.0+1.20.5
 maven_group=me.nahkd
 
 # Fabric Properties
 # check these on https://fabricmc.net/develop
 # Loader 0.15.0 includes MixinExtras, pretty based indeed
-minecraft_version=1.20.3
-yarn_mappings=1.20.3+build.1
-loader_version=0.15.0
+minecraft_version=1.20.5
+yarn_mappings=1.20.5+build.1
+loader_version=0.15.10
 
 # Spigot Plugin Dependencies
 # <Not available yet>
@@ -18,6 +17,6 @@ loader_version=0.15.0
 # Fabric Mod Dependencies
 nahara_toolkit_version=b6ff2e287b
 nahara_modkit_version=d1f7cabf9b
-fabric_version=0.91.1+1.20.3
+fabric_version=0.97.6+1.20.5
 common_economy_version=1.1.1
-sgui_version=1.3.0+1.20.2
+sgui_version=1.5.0+1.20.5

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.3-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.6-bin.zip
 networkTimeout=10000
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists

--- a/settings.gradle
+++ b/settings.gradle
@@ -6,7 +6,7 @@ pluginManagement {
 }
 
 plugins {
-    id 'org.gradle.toolchains.foojay-resolver-convention' version '0.4.0'
+    id 'org.gradle.toolchains.foojay-resolver-convention' version '0.8.0'
 }
 
 include 'core'

